<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Filtering of Features in an ExpressionSet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for nsFilter {genefilter}"><tr><td>nsFilter {genefilter}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Filtering of Features in an ExpressionSet</h2>

<h3>Description</h3>

<p>The function <code>nsFilter</code> tries to provide a one-stop shop for
different options of filtering (removing) features from an ExpressionSet. 
Filtering features exhibiting little variation, or a consistently low
signal, across samples can be advantageous for
the subsequent data analysis (Bourgon et al.).
Furthermore, one may decide that there is little value in considering
features with insufficient annotation.
</p>


<h3>Usage</h3>

<pre>
nsFilter(eset, require.entrez=TRUE,
    require.GOBP=FALSE, require.GOCC=FALSE,
    require.GOMF=FALSE, require.CytoBand=FALSE,
    remove.dupEntrez=TRUE, var.func=IQR,
    var.cutoff=0.5, var.filter=TRUE,
    filterByQuantile=TRUE, feature.exclude="^AFFX", ...)

varFilter(eset, var.func=IQR, var.cutoff=0.5, filterByQuantile=TRUE)

featureFilter(eset, require.entrez=TRUE,
    require.GOBP=FALSE, require.GOCC=FALSE,
    require.GOMF=FALSE, require.CytoBand=FALSE,
    remove.dupEntrez=TRUE, feature.exclude="^AFFX")
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>eset</code></td>
<td>
<p>an <code>ExpressionSet</code> object</p>
</td></tr>
<tr valign="top"><td><code>var.func</code></td>
<td>
<p>The function used as the per-feature filtering
statistic. This function should return a numeric vector of length
one when given a numeric vector as input.</p>
</td></tr>
<tr valign="top"><td><code>var.filter</code></td>
<td>
<p>A logical indicating whether to perform
filtering based on <code>var.func</code>.</p>
</td></tr>  
<tr valign="top"><td><code>filterByQuantile</code></td>
<td>
<p>A logical indicating whether <code>var.cutoff</code>
is to be interprested as a quantile of all <code>var.func</code> values
(the default), or as an absolute value.</p>
</td></tr>
<tr valign="top"><td><code>var.cutoff</code></td>
<td>
<p>A numeric value. If <code>var.filter</code> is TRUE,
features whose value of <code>var.func</code> is less than either:
the <code>var.cutoff</code>-quantile of all <code>var.func</code> values
(if <code>filterByQuantile</code> is TRUE), or
<code>var.cutoff</code> (if <code>filterByQuantile</code> is FALSE)
will be removed.</p>
</td></tr>
<tr valign="top"><td><code>require.entrez</code></td>
<td>
<p>If <code>TRUE</code>, filter out features
without an Entrez Gene ID annotation. If using an annotation
package where an identifier system other than Entrez Gene IDs is
used as the central ID, then that ID will be required instead.</p>
</td></tr>
<tr valign="top"><td><code>require.GOBP, require.GOCC, require.GOMF</code></td>
<td>
<p>If <code>TRUE</code>, filter out features
whose target genes are not annotated to at least one GO term in
the BP, CC or MF ontology, respectively.</p>
</td></tr>
<tr valign="top"><td><code>require.CytoBand</code></td>
<td>
<p>If <code>TRUE</code>, filter out features
whose target genes have no mapping to cytoband locations.</p>
</td></tr> 
<tr valign="top"><td><code>remove.dupEntrez</code></td>
<td>
<p>If <code>TRUE</code> and there are features
mapping to the same Entrez Gene ID (or equivalent), then the feature with
the largest value of <code>var.func</code> will be retained and the
other(s) removed.</p>
</td></tr>
<tr valign="top"><td><code>feature.exclude</code></td>
<td>
<p>A character vector of regular expressions.
Feature identifiers (i.e. value of <code>featureNames(eset)</code>)
that match one of the specified patterns will be filtered out.
The default value is intended to filter out Affymetrix quality control
probe sets.</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>Unused, but available for specializing methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this Section, the effect of filtering on the type I error rate
estimation / control of subsequent hypothesis testing is explained.
See also the paper by Bourgon et al.
</p>
<p><em>Marginal type I errors</em>:
Filtering on the basis of a statistic which is independent of the test
statistic used for detecting differential gene expression can increase
the detection rate at the same marginal type I error. This is
clearly the case for filter criteria that do not depend on the data,
such as the annotation based criteria provided by the <code>nsFilter</code>
and <code>featureFilter</code> functions. However, marginal type I error can
also be controlled for certain types of data-dependent criteria.
Call <i>U^1</i> the stage 1 filter statistic, which is a function
that is applied feature by feature,
based on whose value the feature is or is not accepted to
pass to stage 2, and which depends only on the data for that feature
and not any other feature, and call
<i>U^2</i> the stage 2 test statistic for differential expression.
Sufficient conditions for marginal type-I error control are:
</p>

<ul>
<li> <p><i>U^1</i> the overall (across all samples) variance or
mean, 
<i>U^2</i> the t-statistic (or any other scale and location
invariant statistic),
data normal distributed and exchangeable across samples.
</p>
</li>
<li> <p><i>U^1</i> the overall mean, 
<i>U^2</i> the moderated t-statistic
(as in limma's <code><a href="../../limma/html/ebayes.html">eBayes</a></code> function),
data normal distributed and exchangeable.
</p>
</li>
<li> <p><i>U^1</i> a sample-class label independent function
(e.g. overall mean, median, variance, IQR),
<i>U^2</i> the Wilcoxon rank sum statistic,
data exchangeable.
</p>
</li></ul>

<p><em>Experiment-wide type I error</em>:
Marginal type-I error control provided by the conditions above
is sufficient for control of the family wise error rate (FWER).
Note, however, that common false discovery rate (FDR) methods depend 
not only on the marginal behaviour of the test statistics under the
null hypothesis, but also on their joint distribution.
The joint distribution can be affected by filtering,
even when this filtering leaves the marginal distributions of 
true-null test statistics unchanged. Filtering might, for example, 
change correlation structure. The
effect of this is negligible in many cases in practice, but this
depends on the dataset and the filter used, and the assessment 
is in the responsibility of the data analyst.
</p>
<p><em>Annotation Based Filtering</em> Arguments <code>require.entrez</code>,
<code>require.GOBP</code>, <code>require.GOCC</code>, <code>require.GOMF</code> and
<code>require.CytoBand</code>
filter based on available annotation data.  The annotation
package is determined by calling <code>annotation(eset)</code>.
</p>
<p><em>Variance Based Filtering</em> The <code>var.filter</code>,
<code>var.func</code>, <code>var.cutoff</code> and <code>varByQuantile</code> arguments
control numerical cutoff-based filtering.
Probes for which <code>var.func</code> returns <code>NA</code> are
removed.
The default <code>var.func</code> is <code>IQR</code>, which we here define as
<code>rowQ(eset, ceiling(0.75 * ncol(eset))) - rowQ(eset, floor(0.25 * ncol(eset)))</code>;
this choice is motivated by the observation that unexpressed genes are
detected most reliably through low variability of their features
across samples.
Additionally, <code>IQR</code> is robust to outliers (see note below). The
default <code>var.cutoff</code> is <code>0.5</code> and is motivated by a rule of
thumb that in many tissues only 40% of genes are expressed.
Please adapt this value to your data and question.
</p>
<p>By default the numerical-filter cutoff is interpreted
as a quantile, so with the default settings, 
50% of the genes are filtered. 
</p>
<p>Variance filtering is performed last, so that
(if <code>varByQuantile=TRUE</code> and <code>remove.dupEntrez=TRUE</code>) the
final number of genes does indeed exclude precisely the <code>var.cutoff</code> 
fraction of unique genes remaining after all other filters were
passed.
</p>
<p>The stand-alone function <code>varFilter</code> does only
<code>var.func</code>-based filtering
(and no annotation based filtering).
<code>featureFilter</code> does only
annotation based filtering and duplicate removal; it always
performs duplicate removal to retain the highest-IQR
probe for each gene.
</p>


<h3>Value</h3>

<p>For <code>nsFilter</code> a list consisting of:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>eset</code></td>
<td>
<p>the filtered <code>ExpressionSet</code></p>
</td></tr>
<tr valign="top"><td><code>filter.log</code></td>
<td>
<p>a list giving details of how many probe sets where
removed for each filtering step performed.</p>
</td></tr>
</table>
<p>For both <code>varFilter</code> and <code>featureFilter</code> the filtered
<code>ExpressionSet</code>.
</p>


<h3>Note</h3>

<p><code>IQR</code> is a reasonable variance-filter choice when the dataset
is split into two roughly equal and relatively homogeneous phenotype
groups. If your dataset has important groups smaller than 25% of the
overall sample size, or if you are interested in unusual
individual-level patterns, then <code>IQR</code> may not be sensitive enough
for your needs. In such cases, you should consider using less robust
and more sensitive measures of variance (the simplest of which would
be <code>sd</code>).</p>


<h3>Author(s)</h3>

<p>Seth Falcon (somewhat revised by Assaf Oron)</p>


<h3>References</h3>

<p>R. Bourgon, R. Gentleman, W. Huber,
Independent filtering increases power for detecting differentially
expressed genes, Technical Report.
</p>


<h3>Examples</h3>

<pre>
  library("hgu95av2.db")
  library("Biobase")
  data(sample.ExpressionSet)
  ans &lt;- nsFilter(sample.ExpressionSet)
  ans$eset
  ans$filter.log

  ## skip variance-based filtering
  ans &lt;- nsFilter(sample.ExpressionSet, var.filter=FALSE)

  a1 &lt;- varFilter(sample.ExpressionSet)
  a2 &lt;- featureFilter(sample.ExpressionSet)
</pre>

<hr /><div style="text-align: center;">[Package <em>genefilter</em> version 1.66.0 <a href="00Index.html">Index</a>]</div>
</body></html>
