<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Topic: Analysis of RNA-seq Data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for 11.RNAseq {limma}"><tr><td>11.RNAseq {limma}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Topic: Analysis of RNA-seq Data</h2>

<h3>Description</h3>

<p>This page gives an overview of LIMMA functions to analyze RNA-seq data.
</p>

<dl>
<dt> <code><a href="voom.html">voom</a></code> </dt><dd>
<p>Transform RNA-seq or ChIP-seq counts to log counts per million (log-cpm) with associated precision weights.
After this tranformation, RNA-seq or ChIP-seq data can be analyzed using the same functions as would be used for microarray data.</p>
</dd>
<dt> <code><a href="voomWithQualityWeights.html">voomWithQualityWeights</a></code> </dt><dd>
<p>Combines the functionality of <code>voom</code> and <code>arrayWeights</code>.</p>
</dd>
<dt> <code><a href="diffSplice.html">diffSplice</a></code> </dt><dd>
<p>Test for differential exon usage between experimental conditions.</p>
</dd>
<dt> <code><a href="topSplice.html">topSplice</a></code> </dt><dd>
<p>Show a data.frame of top results from <code>diffSplice</code>.</p>
</dd>
<dt> <code><a href="plotSplice.html">plotSplice</a></code> </dt><dd>
<p>Plot results from <code>diffSplice</code>.</p>
</dd>
<dt> <code><a href="plotExons.html">plotExons</a></code> </dt><dd>
<p>Plot logFC for individual exons for a given gene.</p>
</dd>
</dl>



<h3>References</h3>

<p>Law, CW, Chen, Y, Shi, W, Smyth, GK (2014).
Voom: precision weights unlock linear model analysis tools for RNA-seq read counts.
<em>Genome Biology</em> 15, R29.
<a href="http://genomebiology.com/2014/15/2/R29">http://genomebiology.com/2014/15/2/R29</a>
</p>
<p>Ritchie, ME, Phipson, B, Wu, D, Hu, Y, Law, CW, Shi, W, and Smyth, GK (2015).
limma powers differential expression analyses for RNA-sequencing and microarray studies.
<em>Nucleic Acids Research</em> 43, e47.
<a href="http://nar.oxfordjournals.org/content/43/7/e47">http://nar.oxfordjournals.org/content/43/7/e47</a>
</p>


<h3>See Also</h3>

<p>See also the edgeR package for normalization and data summaries of RNA-seq data, as well as for alternative differential expression methods based on the negative binomial distribution.
<code>voom</code> accepts DGEList objects and normalization factors from edgeR.
</p>
<p><a href="01Introduction.html">01.Introduction</a>,
<a href="02classes.html">02.Classes</a>,
<a href="03reading.html">03.ReadingData</a>,
<a href="04Background.html">04.Background</a>,
<a href="05Normalization.html">05.Normalization</a>,
<a href="06linearmodels.html">06.LinearModels</a>,
<a href="07SingleChannel.html">07.SingleChannel</a>,
<a href="08Tests.html">08.Tests</a>,
<a href="09Diagnostics.html">09.Diagnostics</a>,
<a href="10GeneSetTests.html">10.GeneSetTests</a>,
<a href="11RNAseq.html">11.RNAseq</a>
</p>

<hr /><div style="text-align: center;">[Package <em>limma</em> version 3.40.6 <a href="00Index.html">Index</a>]</div>
</body></html>
