<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Genewise Nested F-Tests</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for classifyTestsF {limma}"><tr><td>classifyTestsF {limma}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Genewise Nested F-Tests</h2>

<h3>Description</h3>

<p>For each gene, classify a series of related t-statistics as significantly up or down using nested F-tests.
</p>


<h3>Usage</h3>

<pre>
classifyTestsF(object, cor.matrix = NULL, df = Inf, p.value = 0.01, fstat.only = FALSE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>object</code></td>
<td>
<p>numeric matrix of t-statistics or an <code>MArrayLM</code> object from which the t-statistics may be extracted.</p>
</td></tr>
<tr valign="top"><td><code>cor.matrix</code></td>
<td>
<p>covariance matrix of each row of t-statistics. Will be extracted automatically from an <code>MArrayLM</code> object but otherwise defaults to the identity matrix.</p>
</td></tr>
<tr valign="top"><td><code>df</code></td>
<td>
<p>numeric vector giving the degrees of freedom for the t-statistics.
May have length 1 or length equal to the number of rows of <code>tstat</code>.
Will be extracted automatically from an <code>MArrayLM</code> object but otherwise default to <code>Inf</code>.</p>
</td></tr>
<tr valign="top"><td><code>p.value</code></td>
<td>
<p>numeric value between 0 and 1 giving the desired size of the test.</p>
</td></tr>
<tr valign="top"><td><code>fstat.only</code></td>
<td>
<p>logical, if <code>TRUE</code> then return the overall F-statistic as for <code>FStat</code> instead of classifying the test results.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>classifyTestsF</code> implements the <code>"nestedF"</code> multiple testing option offered by <code><a href="decideTests.html">decideTests</a></code>.
Users should generally use <code>decideTests</code> rather than calling <code>classifyTests</code> directly because, by itself, <code>classifyTests</code> does not incorporate any multiple testing adjustment across genes.
Instead it simply adjusts for multiple testing across contrasts for each gene individually.
</p>
<p><code>classifyTestsF</code> uses a nested F-test approach giving particular attention to correctly classifying genes that have two or more significant t-statistics, i.e., which are differentially expressed in two or more conditions.
For each row of <code>tstat</code>, the overall F-statistics is constructed from the t-statistics as for <code>FStat</code>.
At least one constrast will be classified as significant if and only if the overall F-statistic is significant.
If the overall F-statistic is significant, then the function makes a best choice as to which t-statistics contributed to this result.
The methodology is based on the principle that any t-statistic should be called significant if the F-test is still significant for that row when all the larger t-statistics are set to the same absolute size as the t-statistic in question.
</p>
<p>Compared to conventional multiple testing methods, the nested F-test approach achieves better consistency between related contrasts.
(For example, if B is judged to be different from C, then at least one of B or C should be different to A.)
The approach was first used by Michaud et al (2008).
The nested F-test approach provides <em>weak</em> control of the family-wise error rate, i.e., it correctly controls the type I error rate of calling any contrast as significant if all the null hypotheses are true.
In other words, it provides error rate control at the overall F-test level but does not provide strict error rate control at the individual contrast level. 
</p>
<p>Usually <code>object</code> is a limma linear model fitted object, from which a matrix of t-statistics can be extracted, but it can also be a numeric matrix of t-statistics.
In either case, rows correspond to genes and columns to coefficients or contrasts.
If <code>object</code> is a matrix, then it may be necessary to supply values for <code>cor.matrix</code> and <code>df</code>.
The <code>cor.matrix</code> is the same as the correlation matrix of the coefficients from which the t-statistics were calculated and <code>df</code> is the degrees of freedom of the t-statistics.
All statistics for the same gene must have the same degrees of freedom.
</p>
<p>If <code>fstat.only=TRUE</code>, the <code>classifyTestsF</code> just returns the vector of overall F-statistics for each gene.
</p>


<h3>Value</h3>

<p>If <code>fstat.only=FALSE</code>, then an object of class <code><a href="TestResults.html">TestResults</a></code> is returned.
This is essentially a numeric matrix with elements <code>-1</code>, <code>0</code> or <code>1</code> depending on whether each t-statistic is classified as significantly negative, not significant or significantly positive respectively.
</p>
<p>If <code>fstat.only=TRUE</code>, then a numeric vector of F-statistics is returned with attributes <code>df1</code> and <code>df2</code> giving the corresponding degrees of freedom.
</p>


<h3>Author(s)</h3>

<p>Gordon Smyth</p>


<h3>References</h3>

<p>Michaud, J, Simpson, KM, Escher, R, Buchet-Poyau, K, Beissbarth, T, Carmichael, C, Ritchie, ME, Schutz, F, Cannon, P, Liu, M, Shen, X, Ito, Y, Raskind, WH, Horwitz, MS, Osato, M, Turner, DR, Speed, TP, Kavallaris, M, Smyth, GK, and Scott, HS (2008).
Integrative analysis of RUNX1 downstream pathways and target genes.
<em>BMC Genomics</em> 9, 363. 
</p>


<h3>See Also</h3>

<p>An overview of multiple testing functions is given in <a href="08Tests.html">08.Tests</a>.
</p>


<h3>Examples</h3>

<pre>
TStat &lt;- matrix(c(0,10,0, 0,5,0, -4,-4,4, 2,2,2), 4, 3, byrow=TRUE)
colnames(TStat) &lt;- paste0("Contrast",1:3)
rownames(TStat) &lt;- paste0("Gene",1:4)
classifyTestsF(TStat, df=20)
FStat &lt;- classifyTestsF(TStat, df=20, fstat.only=TRUE)
P &lt;- pf(FStat, df1=attr(FStat,"df1"), df2=attr(FStat,"df2"), lower.tail=FALSE)
data.frame(F.Statistic=FStat,P.Value=P)
</pre>

<hr /><div style="text-align: center;">[Package <em>limma</em> version 3.40.6 <a href="00Index.html">Index</a>]</div>
</body></html>
