<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: MA-Plot of Expression Data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for plotMA {limma}"><tr><td>plotMA {limma}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>MA-Plot of Expression Data</h2>

<h3>Description</h3>

<p>Creates an MA-plot with color coding for control spots.
</p>


<h3>Usage</h3>

<pre>
## Default S3 method:
plotMA(object, array = 1, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)",
       main = colnames(object)[array], status=NULL, ...)
## S3 method for class 'EList'
plotMA(object, array = 1, xlab = "Average log-expression",
       ylab = "Expression log-ratio (this sample vs others)",
       main = colnames(object)[array], status=object$genes$Status,
       zero.weights = FALSE, ...)
## S3 method for class 'RGList'
plotMA(object, array = 1, xlab = "A", ylab = "M",
       main = colnames(object)[array], status=object$genes$Status,
       zero.weights = FALSE, ...)
## S3 method for class 'MAList'
plotMA(object, array = 1, xlab = "A", ylab = "M",
       main = colnames(object)[array], status=object$genes$Status,
       zero.weights = FALSE, ...)
## S3 method for class 'MArrayLM'
plotMA(object, coef = ncol(object), xlab = "Average log-expression",
       ylab = "log-fold-change", main = colnames(object)[coef],
       status=object$genes$Status, zero.weights = FALSE, ...)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>object</code></td>
<td>
<p>an <code>RGList</code>, <code>MAList</code>, <code>EList</code>, <code>ExpressionSet</code> or <code>MArrayLM</code> object.
Alternatively a numeric <code>matrix</code>.</p>
</td></tr>
<tr valign="top"><td><code>array</code></td>
<td>
<p>integer giving the array to be plotted.</p>
</td></tr>
<tr valign="top"><td><code>coef</code></td>
<td>
<p>integer giving the linear model coefficient to be plotted.</p>
</td></tr>
<tr valign="top"><td><code>xlab</code></td>
<td>
<p>character string, label for x-axis</p>
</td></tr>
<tr valign="top"><td><code>ylab</code></td>
<td>
<p>character string, label for y-axis</p>
</td></tr>
<tr valign="top"><td><code>main</code></td>
<td>
<p>character string, title for plot</p>
</td></tr>
<tr valign="top"><td><code>status</code></td>
<td>
<p>vector giving the control status of each spot on the array, of same length as the number of rows of <code>object</code>.
If <code>NULL</code>, then all points are plotted in the default color, symbol and size.</p>
</td></tr>
<tr valign="top"><td><code>zero.weights</code></td>
<td>
<p>logical, should spots with zero or negative weights be plotted?</p>
</td></tr>
<tr valign="top"><td><code>...</code></td>
<td>
<p>other arguments are passed to <code><a href="plotWithHighlights.html">plotWithHighlights</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An MA-plot is a plot of log-intensity ratios (M-values) versus log-intensity averages (A-values).
See Ritchie et al (2015) for a brief historical review.
</p>
<p>For two color data objects, a within-array MA-plot is produced with the M and A values computed from the two channels for the specified array.
This is the same as a mean-difference plot (<code><a href="mdplot.html">mdplot</a></code>) with the red and green log2-intensities of the array providing the two columns.
</p>
<p>For single channel data objects, a between-array MA-plot is produced.
An artificial array is produced by averaging all the arrays other than the array specified.
A mean-difference plot is then producing from the specified array and the artificial array.
Note that this procedure reduces to an ordinary mean-difference plot when there are just two arrays total.
</p>
<p>If <code>object</code> is an <code>MArrayLM</code> object, then the plot is an fitted model MA-plot in which the estimated coefficient is on the y-axis and the average A-value is on the x-axis.
</p>
<p>The <code>status</code> vector can correspond to any grouping of the probes that is of interest.
If <code>object</code> is a fitted model object, then <code>status</code> vector is often used to indicate statistically significance, so that differentially expressed points are highlighted.
If <code>object</code> is a microarray data object, then <code>status</code> might distinguish control probes from regular probes so that different types of controls are highlighted.
</p>
<p>The <code>status</code> can be included as the component <code>object$genes$Status</code> instead of being passed as an argument to <code>plotMA</code>.
</p>
<p>See <code><a href="plotWithHighlights.html">plotWithHighlights</a></code> for how to set colors and graphics parameters for the highlighted and non-highlighted points.
</p>


<h3>Value</h3>

<p>A plot is created on the current graphics device.</p>


<h3>Note</h3>

<p>The <code><a href="plotMD.html">plotMD</a></code> function provides the same functionality as <code>plotMA</code> with slightly different arguments.
</p>


<h3>Author(s)</h3>

<p>Gordon Smyth</p>


<h3>References</h3>

<p>Ritchie, ME, Phipson, B, Wu, D, Hu, Y, Law, CW, Shi, W, and Smyth, GK (2015).
limma powers differential expression analyses for RNA-sequencing and microarray studies.
<em>Nucleic Acids Research</em> Volume 43, e47.
<a href="http://nar.oxfordjournals.org/content/43/7/e47">http://nar.oxfordjournals.org/content/43/7/e47</a>
</p>


<h3>See Also</h3>

<p>The driver function for <code>plotMA</code> is <code><a href="plotWithHighlights.html">plotWithHighlights</a></code>.
</p>
<p>An overview of plot functions available in LIMMA is given in <a href="09Diagnostics.html">09.Diagnostics</a>.
</p>


<h3>Examples</h3>

<pre>
A &lt;- runif(1000,4,16)
y &lt;- A + matrix(rnorm(1000*3,sd=0.2),1000,3)
status &lt;- rep(c(0,-1,1),c(950,40,10))
y[,1] &lt;- y[,1] + status
plotMA(y, array=1, status=status, values=c(-1,1), hl.col=c("blue","red"))

MA &lt;- new("MAList")
MA$A &lt;- runif(300,4,16)
MA$M &lt;- rt(300,df=3)

# Spike-in values
MA$M[1:3] &lt;- 0
MA$M[4:6] &lt;- 3
MA$M[7:9] &lt;- -3

status &lt;- rep("Gene",300)
status[1:3] &lt;- "M=0"
status[4:6] &lt;- "M=3"
status[7:9] &lt;- "M=-3"
values &lt;- c("M=0","M=3","M=-3")
col &lt;- c("blue","red","green")

plotMA(MA,main="MA-Plot with 12 spiked-in points",
       status=status, values=values, hl.col=col)

#  Same as above but setting graphical parameters as attributes
attr(status,"values") &lt;- values
attr(status,"col") &lt;- col
plotMA(MA, main="MA-Plot with 12 spiked-in points", status=status)

#  Same as above but passing status as part of object
MA$genes$Status &lt;- status
plotMA(MA, main="MA-Plot with 12 spiked-in points")

#  Change settings for background points
MA$genes$Status &lt;- status
plotMA(MA, bg.pch=1, bg.cex=0.5)
</pre>

<hr /><div style="text-align: center;">[Package <em>limma</em> version 3.40.6 <a href="00Index.html">Index</a>]</div>
</body></html>
