<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Read Illumina expression data directly from IDAT files</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for read.idat {limma}"><tr><td>read.idat {limma}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Read Illumina expression data directly from IDAT files</h2>

<h3>Description</h3>

<p>Read Illumina BeadArray data from IDAT and manifest (.bgx) files for gene expression platforms.</p>


<h3>Usage</h3>

<pre>
read.idat(idatfiles, bgxfile, dateinfo = FALSE, annotation = "Symbol",
          tolerance = 0L, verbose = TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>idatfiles</code></td>
<td>
<p>character vector specifying idat files to be read in.</p>
</td></tr>
<tr valign="top"><td><code>bgxfile</code></td>
<td>
<p>character string specifying bead manifest file (.bgx) to be read in.</p>
</td></tr>
<tr valign="top"><td><code>dateinfo</code></td>
<td>
<p>logical. Should date and software version information be read in?</p>
</td></tr>
<tr valign="top"><td><code>annotation</code></td>
<td>
<p>character vector of annotation columns to be read from the manifest file.</p>
</td></tr>
<tr valign="top"><td><code>tolerance</code></td>
<td>
<p>integer. The number of probe ID discrepancies allowed between the manifest and any of the IDAT files.</p>
</td></tr> 
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>logical. Should progress messages are sent to standard output?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Illumina's BeadScan/iScan software outputs probe intensities in IDAT
format (encrypted XML files) and uses probe information stored in a platform specific manifest file (.bgx).
These files can be processed using the low-level functions <code>readIDAT</code> and <code>readBGX</code> 
from the <code>illuminaio</code> package (Smith et al. 2013).
</p>
<p>The <code>read.idat</code> function provides a convenient way to read these files
into R and to store them in an <code>EListRaw-class</code> object.
The function serves a similar purpose to <code><a href="read.ilmn.html">read.ilmn</a></code>, 
which reads text files exported by Illumina's GenomeStudio software,
but it reads the IDAT files directly without any need to convert them first to text.
</p>
<p>The function reads information on control probes as well for regular probes.
Probe types are indicated in the <code>Status</code> column of the <code>genes</code> 
component of the <code>EListRaw</code> object.
</p>
<p>The <code>annotation</code> argument specifies probe annotation columns to be extracted from the manifest file.
The manifest typically contains the following columns:
<code>"Species"</code>, <code>"Source"</code>, <code>"Search_Key"</code>, <code>"Transcript"</code>,
<code>"ILMN_Gene"</code>, <code>"Source_Reference_ID"</code>, <code>"RefSeq_ID"</code>,
<code>"Unigene_ID"</code>, <code>"Entrez_Gene_ID"</code>, <code>"GI"</code>,
<code>"Accession"</code>, <code>"Symbol"</code>, <code>"Protein_Product"</code>,
<code>"Probe_Id"</code>, <code>"Array_Address_Id"</code>, <code>"Probe_Type"</code>,
<code>"Probe_Start"</code>, <code>"Probe_Sequence"</code>, <code>"Chromosome"</code>,
<code>"Probe_Chr_Orientation"</code>, <code>"Probe_Coordinates"</code>, <code>"Cytoband"</code>,
<code>"Definition"</code>, <code>"Ontology_Component"</code>, <code>"Ontology_Process"</code>,
<code>"Ontology_Function"</code>, <code>"Synonyms"</code>, <code>"Obsolete_Probe_Id"</code>.
Note that <code>"Probe_Id"</code> and <code>"Array_Address_Id"</code> are always extracted and
do not need to included in the <code>annotation</code> argument.
</p>
<p>If more than <code>tolerance</code> probes in the manifest cannot be found in an IDAT file then the function will return an error.
</p>


<h3>Value</h3>

<p>An <code>EListRaw</code> object with the following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>E</code></td>
<td>
<p> numeric matrix of raw intensities.</p>
</td></tr>
<tr valign="top"><td><code>other$NumBeads</code></td>
<td>
<p> numeric matrix of same dimensions as <code>E</code> giving number of beads used for each intensity value.</p>
</td></tr>
<tr valign="top"><td><code>other$STDEV</code></td>
<td>
<p> numeric matrix of same dimensions as <code>E</code> giving bead-level standard deviation or standard error for each intensity value.</p>
</td></tr>
<tr valign="top"><td><code>genes</code></td>
<td>
<p> data.frame of probe annotation.
This includes the <code>Probe_Id</code> and <code>Array_Address_Id</code> columns extracted from the manifest file,
plus a <code>Status</code> column identifying control probes,
plus any other columns specified by <code>annotation</code>.</p>
</td></tr>
<tr valign="top"><td><code>targets</code></td>
<td>
<p> data.frame of sample information.
This includes the IDAT file names plus other columns if <code>dateinfo=TRUE</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Matt Ritchie</p>


<h3>References</h3>

<p>Smith ML, Baggerly KA, Bengtsson H, Ritchie ME, Hansen KD (2013). 
illuminaio: An open source IDAT parsing tool. <em>F1000 Research</em> 2, 264.
<a href="http://f1000research.com/articles/2-264/">http://f1000research.com/articles/2-264/</a>
</p>


<h3>See Also</h3>

<p><code><a href="read.ilmn.html">read.ilmn</a></code> imports gene expression data output by GenomeStudio.
</p>
<p><code><a href="nec.html">neqc</a></code> performs normexp by control background correction, log
transformation and quantile between-array normalization for
Illumina expression data.
</p>
<p><code><a href="propexpr.html">propexpr</a></code> estimates the proportion of expressed probes in a microarray.
</p>
<p><code><a href="detectionPValue.html">detectionPValues</a></code> computes detection p-values from the negative controls.
</p>


<h3>Examples</h3>

<pre>
## Not run: 
idatfiles &lt;- dir(pattern="idat")
bgxfile &lt;- dir(pattern="bgx")
x &lt;- read.idat(idatfiles, bgxfile)
x$other$Detection &lt;- detectionPValues(x)
propexpr(data)
y &lt;- neqc(data)

## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>limma</em> version 3.40.6 <a href="00Index.html">Index</a>]</div>
</body></html>
