<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Nested Containment List objects</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for NCList-class {IRanges}"><tr><td>NCList-class {IRanges}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Nested Containment List objects</h2>

<h3>Description</h3>

<p>The NCList class is a container for storing the Nested Containment
List representation of a <a href="IntegerRanges-class.html">IntegerRanges</a> object. Preprocessing a
<a href="IntegerRanges-class.html">IntegerRanges</a> object as a Nested Containment List allows
efficient overlap-based operations like <code><a href="findOverlaps-methods.html">findOverlaps</a></code>.
</p>
<p>The NCLists class is a container for storing a collection of NCList objects.
An NCLists object is typically the result of preprocessing each list
element of a <a href="IntegerRangesList-class.html">IntegerRangesList</a> object as a Nested Containment List.
Like with NCList, the NCLists object can then be used for efficient
overlap-based operations.
</p>
<p>To preprocess a <a href="IntegerRanges-class.html">IntegerRanges</a> or <a href="IntegerRangesList-class.html">IntegerRangesList</a> object,
simply call the <code>NCList</code> or <code>NCLists</code> constructor function on it.
</p>


<h3>Usage</h3>

<pre>
NCList(x, circle.length=NA_integer_)
NCLists(x, circle.length=NA_integer_)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>x</code></td>
<td>

<p>The <a href="IntegerRanges-class.html">IntegerRanges</a> or <a href="IntegerRangesList-class.html">IntegerRangesList</a> object to preprocess.
</p>
</td></tr>
<tr valign="top"><td><code>circle.length</code></td>
<td>

<p>Use only if the space (or spaces if <code>x</code> is a <a href="IntegerRangesList-class.html">IntegerRangesList</a>
object) on top of which the ranges in <code>x</code> are defined needs (need)
to be considered circular. If that's the case, then use
<code>circle.length</code> to specify the length(s) of the circular space(s).
</p>
<p>For <code>NCList</code>, <code>circle.length</code> must be a single positive
integer (or NA if the space is linear).
</p>
<p>For <code>NCLists</code>, it must be an integer vector parallel to <code>x</code>
(i.e. same length) and with positive or NA values (NAs indicate linear
spaces). 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <span class="pkg">GenomicRanges</span> package also defines the
<code><a href="../../GenomicRanges/html/GNCList.html">GNCList</a></code> constructor and class for
preprocessing and representing a vector of genomic ranges as a
data structure based on Nested Containment Lists.
</p>
<p>Some important differences between the new findOverlaps/countOverlaps
implementation based on Nested Containment Lists (BioC &gt;= 3.1) and the
old implementation based on Interval Trees (BioC &lt; 3.1):
</p>

<ul>
<li><p> With the new implementation, the hits returned by
<code><a href="findOverlaps-methods.html">findOverlaps</a></code> are not <em>fully</em> ordered (i.e. ordered
by queryHits and subject Hits) anymore, but only <em>partially</em>
ordered (i.e. ordered by queryHits only). Other than that, and
except for the 2 particular situations mentioned below, the 2
implementations produce the same output. However, the new
implementation is faster and more memory efficient.
</p>
</li>
<li><p> With the new implementation, either the query or the subject can
be preprocessed with <code>NCList</code> for a <a href="IntegerRanges-class.html">IntegerRanges</a>
object (replacement for <code>IntervalTree</code>), <code>NCLists</code>
for a <a href="IntegerRangesList-class.html">IntegerRangesList</a> object (replacement for
<code>IntervalForest</code>), and
<code><a href="../../GenomicRanges/html/GNCList.html">GNCList</a></code> for a
<a href="../../GenomicRanges/html/GenomicRanges.html">GenomicRanges</a> object (replacement for
<code>GIntervalTree</code>).
However, for a one-time use, it is NOT advised to explicitely
preprocess the input. This is because <code><a href="findOverlaps-methods.html">findOverlaps</a></code>
or <code><a href="findOverlaps-methods.html">countOverlaps</a></code> will take care of it and do a better
job at it (by preprocessing only what's needed when it's needed,
and releasing memory as they go).
</p>
</li>
<li><p> With the new implementation, <code><a href="findOverlaps-methods.html">countOverlaps</a></code> on
<a href="IntegerRanges-class.html">IntegerRanges</a> or <a href="../../GenomicRanges/html/GenomicRanges.html">GenomicRanges</a>
objects doesn't call <code><a href="findOverlaps-methods.html">findOverlaps</a></code> in order to
collect all the hits in a growing <a href="../../S4Vectors/html/Hits-class.html">Hits</a> object and count
them only at the end. Instead, the counting happens at the C level
and the hits are not kept. This reduces memory usage considerably
when there is a lot of hits.
</p>
</li>
<li><p> When <code>minoverlap=0</code>, zero-width ranges are now interpreted
as insertion points and considered to overlap with ranges that
contain them. With the old alogrithm, zero-width ranges were always
ignored. This is the 1st situation where the new and old
implementations produce different outputs.
</p>
</li>
<li><p> When using <code>select="arbitrary"</code>, the new implementation will
generally not select the same hits as the old implementation. This is
the 2nd situation where the new and old implementations produce
different outputs.
</p>
</li>
<li><p> The new implementation supports preprocessing of a
<a href="../../GenomicRanges/html/GenomicRanges.html">GenomicRanges</a> object with ranges defined
on circular sequences (e.g. on the mitochnodrial chromosome).
See <a href="../../GenomicRanges/html/GNCList.html">GNCList</a> in the <span class="pkg">GenomicRanges</span>
package for some examples.
</p>
</li>
<li><p> Objects preprocessed with <code>NCList</code>, <code>NCLists</code>, and
<code><a href="../../GenomicRanges/html/GNCList.html">GNCList</a></code> are serializable (with
<code>save</code>) for later use. Not a typical thing to do though,
because preprocessing is very cheap (i.e. very fast and memory
efficient).
</p>
</li></ul>



<h3>Value</h3>

<p>An NCList object for the <code>NCList</code> constructor and an NCLists object
for the <code>NCLists</code> constructor.
</p>


<h3>Author(s)</h3>

<p>Hervé Pagès</p>


<h3>References</h3>

<p>Alexander V. Alekseyenko and Christopher J. Lee &ndash;
Nested Containment List (NCList): a new algorithm for accelerating interval
query of genome alignment and interval databases.
Bioinformatics (2007) 23 (11): 1386-1393.
doi: 10.1093/bioinformatics/btl647
</p>


<h3>See Also</h3>


<ul>
<li><p> The <code><a href="../../GenomicRanges/html/GNCList.html">GNCList</a></code> constructor and class
defined in the <span class="pkg">GenomicRanges</span> package.
</p>
</li>
<li> <p><code><a href="findOverlaps-methods.html">findOverlaps</a></code> for finding/counting interval overlaps
between two <em>range-based</em> objects.
</p>
</li>
<li> <p><a href="IntegerRanges-class.html">IntegerRanges</a> and <a href="IntegerRangesList-class.html">IntegerRangesList</a> objects.
</p>
</li></ul>



<h3>Examples</h3>

<pre>
## The example below is for illustration purpose only and does NOT
## reflect typical usage. This is because, for a one-time use, it is
## NOT advised to explicitely preprocess the input for findOverlaps()
## or countOverlaps(). These functions will take care of it and do a
## better job at it (by preprocessing only what's needed when it's
## needed, and release memory as they go).

query &lt;- IRanges(c(1, 4, 9), c(5, 7, 10))
subject &lt;- IRanges(c(2, 2, 10), c(2, 3, 12))

## Either the query or the subject of findOverlaps() can be preprocessed:

ppsubject &lt;- NCList(subject)
hits1 &lt;- findOverlaps(query, ppsubject)
hits1

ppquery &lt;- NCList(query)
hits2 &lt;- findOverlaps(ppquery, subject)
hits2

## Note that 'hits1' and 'hits2' contain the same hits but not in the
## same order.
stopifnot(identical(sort(hits1), sort(hits2)))
</pre>

<hr /><div style="text-align: center;">[Package <em>IRanges</em> version 2.18.3 <a href="00Index.html">Index</a>]</div>
</body></html>
